### PR TITLE
RBAC changes for globalDNS and multiclusterapp

### PR DESCRIPTION
**Pull request by mrajashree**
1.Change member access types to match project/cluster roles: owner,member,read-only
2.Make projectIDs non updatable

https://github.com/rancher/rancher/issues/17564
https://github.com/rancher/rancher/issues/17565

### CODE REVIEW

File: apis/management.cattle.io/v3/multi_cluster_app.go [hunk 1]
```diff
@@ -80,3 +81,8 @@ type MultiClusterAppRevision struct {
 type MultiClusterAppRollbackInput struct {
 	RevisionName string `json:"revisionName,omitempty" norman:"type=reference[multiClusterAppRevision]"`
 }
+
+type UpdateMultiClusterAppTargetsInput struct {
+	Projects []string `json:"projects" norman:"type=array[reference[project]],required"`
+	Answers  []Answer `json:"answers" norman:"type=array[reference[answer]]"`
+}
```
Comment (prachidamle) [inline apis/management.cattle.io/v3/multi_cluster_app.go]: Should not this accept []Target? We seem to miss the appName/Healthstate from Targets?

Comment (mrajashree) [reply apis/management.cattle.io/v3/multi_cluster_app.go]: App name is assigned by the controllers when apps are created for those targets, and so should healthstate eventually when we have logic for it

File: apis/management.cattle.io/v3/globaldns_types.go [hunk 2]
```diff
@@ -57,3 +57,11 @@ type CloudflareProviderConfig struct {
 	APIKey     string `json:"apiKey" norman:"notnullable,required,minLength=1,type=password"`
 	APIEmail   string `json:"apiEmail" norman:"notnullable,required,minLength=1"`
 }
+
+type UpdateTargetsInput struct {
+	Projects []string `json:"projects" norman:"type=array[reference[project]]"`
```
[the remark this answers is not in the document]
Comment (mrajashree) [reply apis/management.cattle.io/v3/globaldns_types.go]: I think this is fine too?

**Comment by prachidamle** [comment]
LGTM

**Pull request merged**